libs/ui-lib/lib/ocm/components/clusterWizard/CredentialsDownload.tsx

```diff
@@ -0,0 +1,137 @@
+import React from 'react';
+import {
+  ClusterCredentials,
+  ClustersAPI,
+  ClusterWizardStep,
+  ClusterWizardStepHeader,
+  getApiErrorMessage,
+  handleApiError,
+  KUBECONFIG_INFO_LINK,
+  useAlerts,
+  useTranslation,
+} from '../../../common';
+import { useClusterWizardContext } from './ClusterWizardContext';
+import ClusterWizardFooter from './ClusterWizardFooter';
+import ClusterWizardNavigation from './ClusterWizardNavigation';
+import { WithErrorBoundary } from '../../../common/components/ErrorHandling/WithErrorBoundary';
+import { Cluster, Credentials } from '@openshift-assisted/types/assisted-installer-service';
+import { Alert, AlertVariant, Checkbox, Grid } from '@patternfly/react-core';
+import { downloadFile, getErrorMessage } from '../../../common/utils';
+import { ExternalLinkAltIcon } from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+
+const CredentialsDownload: React.FC<{ cluster: Cluster }> = ({ cluster }) => {
+  const [isChecked, setIsChecked] = React.useState<boolean>(false);
+  const clusterWizardContext = useClusterWizardContext();
+  const [credentials, setCredentials] = React.useState<Credentials>();
+  const [credentialsError, setCredentialsError] = React.useState('');
```

Comment:
lets use undefined as default
```suggestion
  const [credentialsError, setCredentialsError] = React.useState<string>();
```